reg/drone/README.md

```diff
@@ -0,0 +1,56 @@
+# Drone
+
+This namespace contains the application-specific regulated DSDL namespace for aerial vehicles.
+If you have any questions, feel free to bring them to the [UAVCAN Forum](https://forum.uavcan.org/c/sig/drone-sig/17).
+
+This namespace contains two nested namespaces:
+
+- `physics` -- abstract physical processes and states in the system.
+- `service` -- narrowly specialized types for common device classes.
+
+The motivation for such design is provided in [The UAVCAN Guide](https://uavcan.org/guide),
+chapter "Interface Design Guidelines".
+
+The value of this or any other interoperability standard is in enabling compatible, composable, and extensible
+complex systems.
+A design that does not put these capabilities above the resource utilization concerns would defeat the purpose
+of the standard.
+Hence, when designing a new network service, put the clarity of your models first, then think about their performance
+implications.
+
+## Composable services 101
+
+Make sure you understand the basic principles of service-oriented design.
+As a quick primer, read Wikipedia: https://en.wikipedia.org/wiki/Service-oriented_architecture.
+For a more in-depth review, please read *The UAVCAN Guide*.
+
+The concrete service definitions given in the `service` namespace are intended for service implementers
+in the first place.
+If your objective is to depend on a service, then focus your attention on the other namespaces,
+because they provide a generalized view of common processes that is invariant to the means of their implementation.
+
+A real hardware node would typically implement multiple services concurrently.
+For example, a COTS ESC may realistically implement the following:
+
+- Naturally, the ESC service.
+- The servo service for generality.
+- Acoustic feedback by subscribing to `reg.drone.physics.acoustics.Node`.
+- Visual feedback via the LED by subscribing to `reg.drone.optics.HighColor`.
+
+Another service that is interested in tracking the state of, say, a propeller drive
+(say, for thrust estimation) would not need to concern itself with the ESC service at all.
+Instead, it would simply subscribe to the generalized subject of type
+`reg.drone.physics.kinetics.Angular1DTs` published by the unit that drives the propeller
+and extract its business-level information from that while being unaware of the specifics of the drive
+(the propeller drive may be changed from an electric motor to a turboprop engine without affecting the
+thrust estimation service).
+
+As another example, a flight control unit would not need to depend on the specifics of a GNSS positioning
+service to obtain the location of the vehicle.
+Instead, it would subscribe to a generic subject of a highly abstract type that models the location of
+the vehicle in space (along with other related information such as time and pose),
+which may as well be published by a mocap rig.
```

Comment:
We need to make clear what is the frame of reference. Mocap and GPS have different frames of reference for example. So that concept needs to exist.

Reply:
Sure, but this is an abstract example in the README, not a specification. I don't think it needs extending.

Reply:
I am just saying that the concept of frame of reference should be part of the data types.

Reply:
A given location service could provide a vector of Linear1DTs and an EPSG Coordinate Reference System code. Subscribers could re-project into a local coordinate system if needed. Could add the concept of a EPSG coord system when a Nav system service is added?

Reply:
The current plan is to define a collection of types describing 3D kinematic states in Cartesian and geoid systems. A rough approximation that I have drafted up so far (it is not public and is very far from being finished) looks like this:

```
kinematics
├── geo
│   ├── Point.0.1.uavcan
│   ├── PointCov.0.1.uavcan
│   ├── Pose.0.1.uavcan
│   ├── PoseCov.0.1.uavcan
│   ├── PoseTwistCov.0.1.uavcan
│   ├── PoseTwistCovTs.0.1.uavcan
│   ├── PositionVelocityCov.0.1.uavcan
│   ├── PositionVelocityCovTs.0.1.uavcan
│   └── VelocityCov.0.1.uavcan
├── Twist.0.1.uavcan
└── TwistCov.0.1.uavcan
```

We will need to agree on the specifics with Auterion but I suspect that we will be working on this shortly after the DS-015 MVP is out and ready.

This conversation is out of the scope of this PR. If you want to influence the future development of this standard, please create a new thread on the forum here: https://forum.uavcan.org/c/sig/drone-sig/17